app/src/main/java/com/sun/cookbook/screen/detail/ingredient/IngredientAdapter.kt

```diff
@@ -0,0 +1,33 @@
+package com.sun.cookbook.screen.detail.ingredient
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.sun.cookbook.R
+import com.sun.cookbook.data.model.Ingredient
+
+class IngredientAdapter(private val onItemClicked: (Ingredient) -> Unit) :
+    RecyclerView.Adapter<ItemIngredientViewHolder>() {
+
+    private val ingredient = arrayListOf<Ingredient>()
+
+    fun updateData(ingredients: ArrayList<Ingredient>) {
+        ingredients.let {
+            this.ingredient.clear()
+            this.ingredient.addAll(ingredients)
+            this.notifyDataSetChanged()
```

Comment:
this có chức năng gì e?

Reply:
> this có chức năng gì e?

Em sợ lộn giữa 2 cái array nên em thêm this. đó anh
mà em có thêm "s" vào cái array trong pram rồi nên this này không có tác dụng mấy :D :D